src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -107,25 +107,27 @@ private function preRender(MountedComponent $mounted, array $context = []): PreR
     {
         $component = $mounted->getComponent();
         $metadata = $this->factory->metadataFor($mounted->getName());
+        $classProps = $metadata->isPublicPropsExposed() ? iterator_to_array($this->exposedVariables($component, $metadata->isPublicPropsExposed())) : [];
+
         // expose public properties and properties marked with ExposeInTemplate attribute
-        $props = iterator_to_array($this->exposedVariables($component, $metadata->isPublicPropsExposed()));
+        $props = array_merge($mounted->getInputProps(), $classProps);
         $variables = array_merge(
             // first so values can be overridden
             $context,
-
+            // add the context in a separate variable to keep track
+            // of what is coming from outside the component
+            ['__context' => $context],
             // keep reference to old context
             ['outerScope' => $context],
-
             // add the component as "this"
             ['this' => $component],
-
             // add computed properties proxy
             ['computed' => new ComputedPropertiesProxy($component)],
-
+            $props,
+            // keep this line for BC break reasons
+            ['__props' => $classProps],
```

Comment:
Nice.. and after this PR maybe we can think of a way to "deprecate" it more clearly